terraform crashed with panic

Hi,

I have faced the problem as below.

### Terraform Version

```
# terraform -v
Terraform v0.12.18
+ provider.datadog v2.14.0

Your version of Terraform is out of date! The latest version
is 0.13.5. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)

-   datadog_synthetics_test

### Output

```
Error: rpc error: code = Unavailable desc = transport is closing


   7: resource "datadog_synthetics_test" "this" {

(and 2 more similar warnings elsewhere)


Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x115175b]
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: goroutine 62 [running]:
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: github.com/terraform-providers/terraform-provider-datadog/datadog.updateSyntheticsTestLocalState(0xc0001d0070, 0xc000b75810, 0xc000a05500, 0xc00092f348)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_synthetics_test_.go:1022 +0x5fb
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogSyntheticsTestRead(0xc0001d0070, 0x120e060, 0xc000204cc0, 0xc0001d0070, 0x0)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_synthetics_test_.go:521 +0x36e
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000522b40, 0xc000ed2be0, 0x120e060, 0xc000204cc0, 0xc00000e550, 0x0, 0x0)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/helper/schema/resource.go:460 +0x129
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc00020c068, 0x167aac0, 0xc000e4e6f0, 0xc00012eae0, 0xc00020c068, 0xc000e4e6f0, 0xc000b34b78)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/internal/helper/plugin/grpc_provider.go:525 +0x3dd
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x1449220, 0xc00020c068, 0x167aac0, 0xc000e4e6f0, 0xc00012ea80, 0x0, 0x167aac0, 0xc000e4e6f0, 0xc000ef4780, 0x27b)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/internal/tfplugin5/tfplugin5.pb.go:3269 +0x214
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000244f00, 0x1684220, 0xc0006c2900, 0xc0001c2100, 0xc0004e36b0, 0x1f90350, 0x0, 0x0, 0x0)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	google.golang.org/grpc@v1.27.1/server.go:1024 +0x522
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: google.golang.org/grpc.(*Server).handleStream(0xc000244f00, 0x1684220, 0xc0006c2900, 0xc0001c2100, 0x0)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	google.golang.org/grpc@v1.27.1/server.go:1313 +0xd34
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000200320, 0xc000244f00, 0x1684220, 0xc0006c2900, 0xc0001c2100)
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	google.golang.org/grpc@v1.27.1/server.go:722 +0xa5
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-10-30T03:55:57.730Z [DEBUG] plugin.terraform-provider-datadog_v2.14.0: 	google.golang.org/grpc@v1.27.1/server.go:720 +0xa5
2020-10-30T03:55:57.733Z [DEBUG] plugin: plugin process exited: path=/root/project/environments/production/synthetics/.terraform/plugins/linux_amd64/terraform-provider-datadog_v2.14.0 pid=717 error="exit status 2"
2020/10/30 03:55:57 [ERROR] module.yyyyy: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [ERROR] module.xxxxx: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [ERROR] module.zzzzz: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [ERROR] module.xxxxx: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [ERROR] module.zzzzz: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [TRACE] [walkRefresh] Exiting eval tree: module.zzzzz.datadog_synthetics_test.this
2020/10/30 03:55:57 [TRACE] [walkRefresh] Exiting eval tree: module.xxxxx.datadog_synthetics_test.this
2020/10/30 03:55:57 [TRACE] vertex "module.zzzzz.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [TRACE] vertex "module.xxxxx.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [ERROR] module.yyyyy: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/10/30 03:55:57 [TRACE] vertex "module.zzzzz.datadog_synthetics_test.this": dynamic subgraph encountered errors
2020/10/30 03:55:57 [TRACE] vertex "module.xxxxx.datadog_synthetics_test.this": dynamic subgraph encountered errors
2020/10/30 03:55:57 [TRACE] vertex "module.zzzzz.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [TRACE] vertex "module.xxxxx.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [TRACE] [walkRefresh] Exiting eval tree: module.yyyyy.datadog_synthetics_test.this
2020/10/30 03:55:57 [TRACE] vertex "module.yyyyy.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [TRACE] vertex "module.yyyyy.datadog_synthetics_test.this": dynamic subgraph encountered errors
2020/10/30 03:55:57 [TRACE] vertex "module.xxxxx.datadog_synthetics_test.this": visit complete
2020/10/30 03:55:57 [TRACE] dag/walk: upstream of "provider.datadog (close)" errored, so skipping
2020/10/30 03:55:57 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-10-30T03:55:57.734Z [DEBUG] plugin: plugin exited
```

### Steps to Reproduce

- terraform plan

--- a/datadog/resource_datadog_synthetics_test_.go
+++ b/datadog/resource_datadog_synthetics_test_.go
@@ -904,7 +904,8 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 		d.Set("subtype", syntheticsTest.GetSubtype())
 	}
 
-	actualRequest := syntheticsTest.GetConfig().Request
+	config := syntheticsTest.GetConfig()
+	actualRequest := config.GetRequest()
 	localRequest := make(map[string]string)
 	if actualRequest.HasBody() {
 		localRequest["body"] = actualRequest.GetBody()
@@ -963,7 +964,7 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 		d.Set("request_client_certificate", []map[string][]map[string]string{localCertificate})
 	}
 
-	actualAssertions := syntheticsTest.GetConfig().Assertions
+	actualAssertions := config.GetAssertions()
 	localAssertions := make([]map[string]interface{}, len(actualAssertions))
 	for i, assertion := range actualAssertions {
 		localAssertion := make(map[string]interface{})
@@ -1019,7 +1020,7 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 		}
 	}
 
-	actualVariables := *syntheticsTest.GetConfig().Variables
+	actualVariables := config.GetVariables()
 	localVariables := make([]map[string]interface{}, len(actualVariables))
 	for i, variable := range actualVariables {
 		localVariable := make(map[string]interface{})

